### DISCUSSION TITLE
Can '--smart-case' apply for every item in '-e' or file specified by '-f'?

### DISCUSSION THREAD

**Discussion by xaljer** [BurntSushi/ripgrep#1787] [General] [answered]
When I use `-S` with `-f` or multiple `-e`, it seems like if only one search pattern in them has upper case (i.e. wanna case sensitive), then all patterns are searched as case sensitive.
For example, `rg -S -e Abc -e bcd` search pattern `bcd` also as case sensitive. For every line in the file specified by `-f`, it's same.
Should this behave be improved to judge every pattern if it need case sensitive?
I think it should be changed especially in the `-f` usage, since I instinctively think it should has same behaves with searching every pattern respectively.

**Comment by BurntSushi** [comment, marked as answer]
This is a plausible thing to do. I thought about this a bit this morning before responding, and I can't think of any good reason not to do this, other than the fact that it might be a bit tricky to pull off in the existing implementation. So doing this might require a bit of refactoring. But it does seem to me like since `--smart-case` is specifically determined by the characteristics of a pattern, then it stands to reason that it should probably apply to each individual pattern.

**Comment by BurntSushi** [comment]
@xaljer Do you want to file a feature request for this? I'd convert this discussion into it, but GitHub doesn't allow it. In the feature request, if possible, please include the added documentation you'd like to see to ripgrep's man page for this change.

**Comment by xaljer** [comment]
@BurntSushi  I opened an issue of feature request about this and added document description in my view. But I'm not native English speaker, I'm not sure if I state it clearly.
